packages/core/src/test-utils/mock-tool.ts

```diff
@@ -37,6 +46,24 @@ interface MockToolOptions {
   params?: object;
 }
 
+export interface MockModifiableToolShouldConfirmExecuteParams {
+  filePath: string;
+  currentContent: string;
+  proposedContent: string;
+}
+
+// disallow shouldConfirmExecute in favor of passing shouldConfirmExecuteParams
+// to populate fields in return of getModifyContext
+interface MockModifiableToolOptions
+  extends Partial<Omit<MockToolOptions, 'shouldConfirmExecute'>> {
+  shouldConfirmExecuteParams: MockModifiableToolShouldConfirmExecuteParams;
+  createUpdatedParamsFn: (
```

Comment:
Heed my comment above, but I think this func should take an options bag RATHER than a bunch of strings.